python/pylibvw.cc

```diff
@@ -857,6 +871,49 @@ uint32_t ex_get_cbandits_class(example_ptr ec, uint32_t i) { return ec->l.cb.cos
 float ex_get_cbandits_probability(example_ptr ec, uint32_t i) { return ec->l.cb.costs[i].probability; }
 float ex_get_cbandits_partial_prediction(example_ptr ec, uint32_t i) { return ec->l.cb.costs[i].partial_prediction; }
 
+const size_t ex_get_ccb_type(example_ptr ec)
+{
+  switch(ec->l.conditional_contextual_bandit.type)
+  {
+    case CCB::example_type::shared:
+      return tSHARED;
+    case CCB::example_type::action:
+      return tACTION;
+    case CCB::example_type::slot:
+      return tSLOT;
+    default:
+      return tUNSET;
+  }
+}
+float ex_get_ccb_cost(example_ptr ec)
+{
+  auto* outcome_ptr = ec->l.conditional_contextual_bandit.conditional_contextual_bandit_outcome;
+  return outcome_ptr == nullptr ? Py_None : outcome_ptr->cost;
+}
+size_t ex_get_ccb_class(example_ptr ec, uint32_t i)
+{
+  auto* outcome_ptr = ec->l.conditional_contextual_bandit.conditional_contextual_bandit_outcome;
+  return outcome_ptr == nullptr ? Py_None : outcome_ptr->probabilities[i]->action;
+}
+float ex_get_ccb_probability(example_ptr ec, uint32_t i)
+{
+  auto* outcome_ptr = ec->l.conditional_contextual_bandit.conditional_contextual_bandit_outcome;
+  return outcome_ptr == nullptr ? Py_None : outcome_ptr->probabilities[i]->score;
+}
+float ex_get_ccb_weight(example_ptr ec) { return ec->l.conditional_contextual_bandit.weight; }
+uint32_t ex_get_ccb_num_included_actions(example_ptr ec)
+{
+  auto label = ec->l.conditional_contextual_bandit;
```

Comment:
This is currently making a copy
```suggestion
  const auto& label = ec->l.conditional_contextual_bandit;
```